pkg/frontend/admin_vm_actions.go

```diff
@@ -0,0 +1,80 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+type VMContext struct {
+	context context.Context
+	r       *http.Request
+	logger  *logrus.Entry
+}
+
+type AzActionsFactoryWrap struct {
+	vmName       string
+	adminActions adminactions.AzureActions
+	doc          *api.OpenShiftClusterDocument
+	vars         map[string]string
+}
+
+func (f *frontend) GetVMContext(r *http.Request) VMContext {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	return VMContext{
+		context: ctx,
+		logger:  log,
+		r:       r,
+	}
+}
+
+func (f *frontend) BuildAzureActionsFactory(ctx context.Context, r *http.Request, log *logrus.Entry) (AzActionsFactoryWrap, error) {
```

Comment:
Doesn't need to be exported.  
```suggestion 
func (f *frontend) buildAzureActionsFactory(ctx context.Context, r *http.Request, log *logrus.Entry) (AzActionsFactoryWrap, error) {
```